### DISCUSSION TITLE
Accessing the underlying Wayland connection

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [hecrj/iced#1826] [Q&A]
Given an `iced::application::Application`, can I somehow get a reference of the underlying Wayland connection (e.g.: the type exposed by `wayland_client`?

I want to interact with the compositor directly to use features/protocols for which iced has no abstractions, but I need to use the same connection.

**Comment by hecrj** [comment]
You cannot currently, but I am open to exposing internals once the `advanced-text` branch lands.

This branch has a new `advanced` feature flag that enables an `advanced` module where this kind of internals could be exposed.

**Reply by WhyNotHugo** [reply]
This explains why I couldn't find how to do it right now :)

Is there an issue/PR that I can follow to track this?

**Reply by hecrj** [reply]
Maybe https://github.com/iced-rs/iced/issues/576

**Reply by hecrj** [reply]
It'd be great to know more about these features you want to use. They may warrant first-class support.

**Reply by WhyNotHugo** [reply]
I mostly want to interact with the clipboard, e.g.: https://github.com/iced-rs/iced/discussions/1455

This definitely warrants first-class support, but I'd rather experiment externally (e.g.: in a more "plugin/extension" mindset) and later evaluate how to integrate it further.

**Reply by WhyNotHugo** [reply]
(FWIW, I've also looked for the same feature in the past with the intent of using layer shell, which has also been brought up in https://github.com/iced-rs/iced/discussions/1214)
